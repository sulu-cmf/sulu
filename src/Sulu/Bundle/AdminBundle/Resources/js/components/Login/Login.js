// @flow
import React from 'react';
import type {ViewProps} from '../../containers/ViewRenderer';
import Icon from '../../components/Icon';
import Input from '../../components/Input';
import Button from '../../components/Button';
import {translate} from '../../utils';
import loginStyles from './login.scss';
import logo from './logo.svg';

const BACK_BUTTON_ARROW_LEFT_ICON = 'chevron-left';

export default class Login extends React.PureComponent<ViewProps> {
    render() {
        return (
            <div className={loginStyles.login}>
                <div className={loginStyles.loginContainer}>
                    <a href="/">
                        <Icon name={BACK_BUTTON_ARROW_LEFT_ICON} />
                        {translate('sulu_admin.back_to_website')}
                    </a>
                    <img src={logo} />
                    <form className={loginStyles.form}>
                        <Input />
                        <Input />
                        <Button />
                    </form>
                </div>
            </div>
        );
    }
}
