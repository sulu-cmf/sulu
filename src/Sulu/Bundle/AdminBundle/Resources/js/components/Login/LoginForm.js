// @flow
import React from 'react';
import {action, observable} from 'mobx';
import {observer} from 'mobx-react';
import classNames from 'classnames';
import {translate} from '../../utils';
import Input from '../../components/Input';
import loginFormStyles from './loginForm.scss';

type Props = {
    onLogin: (user: string, password: string) => void,
    onResetPassword: (user: string) => void,
};

@observer
export default class LoginForm extends React.PureComponent<Props> {
    @observable visibleForm: 'login' | 'reset' = 'login';

    @observable user: string;

    @observable password: string;

    @action handleChangeToLoginForm = () => {
        this.visibleForm = 'login';
    };

    @action handleChangeToResetForm = () => {
        this.visibleForm = 'reset';
    };

    @action handleUserChange = (user: string) => {
        this.user = user;
    };

    @action handlePasswordChange = (password: string) => {
        this.password = password;
    };

    handleLoginFormSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
        this.props.onLogin(this.user, this.password);
        event.preventDefault();
    };

    handleResetFormSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
        this.props.onResetPassword(this.user);
        event.preventDefault();
    };

    render() {
        const loginFormClass = classNames(
            loginFormStyles.form,
            loginFormStyles.loginForm,
            {
                [loginFormStyles.visible]: this.visibleForm === 'login',
            }
        );
        const resetFormClass = classNames(
            loginFormStyles.form,
            loginFormStyles.resetForm,
            {
                [loginFormStyles.visible]: this.visibleForm === 'reset',
            }
        );

        return (
            <div>
                <form className={loginFormClass} onSubmit={this.handleLoginFormSubmit}>
                    <label className={loginFormStyles.inputField}>
                        <div className={loginFormStyles.labelText}>
                            {translate('sulu_admin.username_or_email')}
                        </div>
                        <Input value={this.user} onChange={this.handleUserChange} />
                    </label>
                    <label className={loginFormStyles.inputField}>
                        <div className={loginFormStyles.labelText}>
                            {translate('sulu_admin.password')}
                        </div>
                        <Input value={this.password} onChange={this.handlePasswordChange} />
                        <button
                            type="button"
                            className={loginFormStyles.changeFormButton}
                            onClick={this.handleChangeToResetForm}
                        >
                            {translate('sulu_admin.forgot_password')}
                        </button>
                    </label>
                    <button className={loginFormStyles.submit}>{translate('sulu_admin.login')}</button>
                </form>
                <form className={resetFormClass} onSubmit={this.handleResetFormSubmit}>
                    <label className={loginFormStyles.inputField}>
                        <div className={loginFormStyles.labelText}>
                            {translate('sulu_admin.username_or_email')}
                        </div>
                        <Input value={this.user} onChange={this.handleUserChange} />
                        <button
                            type="button"
                            className={loginFormStyles.changeFormButton}
                            onClick={this.handleChangeToLoginForm}
                        >
                            {translate('sulu_admin.to_login')}
                        </button>
                    </label>
                    <button className={loginFormStyles.submit}>{translate('sulu_admin.reset')}</button>
                </form>
            </div>
        );
    }
}
