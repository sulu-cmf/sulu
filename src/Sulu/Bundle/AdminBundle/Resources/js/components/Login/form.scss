@import '../../containers/Application/colors.scss';

$formTextColor: $white;
$submitBackgroundColor: $shakespeare;
$errorColor: $roman;

.form {
    width: 100%;

    fieldset {
        padding: 0;
        border: 0;
    }
}

.buttons {
    display: flex;
    justify-content: space-between;
    margin-top: 30px;
    width: 100%;
}

.input-field {
    display: block;
    margin-bottom: 10px;

    &.error {
        color: $errorColor;
    }
}

.label-text {
    margin-bottom: 10px;
}
