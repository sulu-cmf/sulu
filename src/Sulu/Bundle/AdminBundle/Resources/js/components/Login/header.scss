.header {
    font-size: 22px;
    font-weight: bold;
    line-height: 32px;
    margin: 10px 0 80px;
    text-align: center;
    width: 100%;
    min-height: 50px;

    &.small {
        font-size: 14px;
        line-height: 20px;
        font-weight: normal;
    }
}
