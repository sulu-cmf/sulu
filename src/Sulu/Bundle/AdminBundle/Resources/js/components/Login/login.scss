@import '../../containers/Application/colors.scss';

$loginTextColor: $white;
$formContainerBackgroundColor: rgba($black, .7);

.login {
    height: 100vh;
    background-image: url('./background.jpg');
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
    color: $loginTextColor;
    border-radius: 3px;
    font-size: 14px;
}

.logo {
    display: block;
    margin: 100px auto 120px;
}

.back-link {
    color: $loginTextColor;
    text-decoration: none;
    display: flex;
    align-items: center;
}

.back-link-icon {
    margin-right: 10px;
}

.form-container {
    background-color: $formContainerBackgroundColor;
    padding: 40px;
    width: 500px;
    height: 680px;
}
