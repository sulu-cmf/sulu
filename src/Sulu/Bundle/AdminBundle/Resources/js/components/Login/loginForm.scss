@import '../../containers/Application/colors.scss';

$formTextColor: $white;
$submitBackgroundColor: $shakespeare;

.form {
    width: 260px;
    margin: 0 auto;
    opacity: 0;
    pointer-events: none;
    transition: all .25s ease-in-out;

    &.visible {
        opacity: 1;
        pointer-events: auto;
    }
}

.login-form {
    transform: translateY(-20px);

    &.visible {
        transform: translateY(0);
    }
}

.reset-form {
    transform: translateY(-150px);

    &.visible {
        transform: translateY(-170px);
    }
}

.input-field {
    display: block;
    margin-bottom: 20px;
}

.label-text {
    margin-bottom: 10px;
}

.submit {
    width: 100%;
    height: 30px;
    color: $formTextColor;
    background-color: $submitBackgroundColor;
    border: 0;
    border-radius: 3px;
    cursor: pointer;
    margin-top: 10px;
}

.change-form-button {
    background-color: transparent;
    border: 0;
    color: $formTextColor;
    margin-top: 10px;
    font-size: 10px;
    padding: 0;
    cursor: pointer;
}
