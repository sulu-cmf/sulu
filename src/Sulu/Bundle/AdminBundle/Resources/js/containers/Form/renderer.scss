@import '../../components/Grid/variables.scss';
@import './variables.scss';

.grid {
    margin-bottom: -$sectionMarginBottom;
}

.grid-section {
    margin-bottom: -$fieldMarginBottom;

    & .grid-section:last-child {
        margin-bottom: 0;
    }

    & + .grid-item {
        margin-top: $fieldMarginBottom;
    }
}

.grid-item + .grid-section {
    margin-top: -$fieldMarginBottom;
}

@media (max-width: $gridMinWidth) {
    .grid-section + .grid-section {
        margin-top: $fieldMarginBottom;
    }
}
