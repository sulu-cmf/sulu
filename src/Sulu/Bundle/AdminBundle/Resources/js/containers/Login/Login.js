// @flow
import React from 'react';
import {action, computed, observable} from 'mobx';
import {observer} from 'mobx-react';
import Icon from '../../components/Icon/index';
import {translate} from '../../utils/index';
import Loader from '../../components/Loader/Loader';
import userStore from '../../stores/UserStore';
import LoginForm from './LoginForm';
import ResetForm from './ResetForm';
import loginStyles from './login.scss';

const BACK_LINK_ARROW_LEFT_ICON = 'su-angle-left';

type Props = {
    backLink: string,
    onLoginSuccess: () => void,
    initialized: boolean,
};

@observer
export default class Login extends React.Component<Props> {
    static defaultProps = {
        backLink: '/',
        initialized: false,
    };

    @observable visibleForm: 'login' | 'reset' = 'login';
    @observable user: ?string;
    @observable password: ?string;

    @computed get loginFormVisible(): boolean {
        return this.visibleForm === 'login';
    }

    @computed get resetFormVisible(): boolean {
        return this.visibleForm === 'reset';
    }

    @action clearState = () => {
        if (this.loginFormVisible) {
            userStore.setLoginError(false);
        } else if (this.resetFormVisible) {
            userStore.setResetSuccess(false);
        }
    };

    @action handleChangeToLoginForm = () => {
        this.visibleForm = 'login';
    };

    @action handleChangeToResetForm = () => {
        this.visibleForm = 'reset';
    };

    @action handleUserChange = (user: ?string) => {
        this.clearState();
        this.user = user;
    };

    @action handlePasswordChange = (password: ?string) => {
        this.clearState();
        this.password = password;
    };

    handleLoginFormSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
        event.preventDefault();

        if (!this.user || !this.password) {
            return;
        }

        userStore.login(this.user, this.password).then(() => {
            this.props.onLoginSuccess();
        });
    };

    handleResetFormSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
        event.preventDefault();

        if (!this.user) {
            return;
        }

        userStore.resetPassword(this.user);
    };

    renderForm() {
        if (!this.props.initialized) {
            return (
                <div className={loginStyles.loaderContainer}>
                    <Loader size={20} />
                </div>
            );
        }

        if (this.loginFormVisible) {
            return (
                <LoginForm
                    error={userStore.loginError}
                    loading={userStore.loading}
                    onChangeForm={this.handleChangeToResetForm}
                    onPasswordChange={this.handlePasswordChange}
                    onSubmit={this.handleLoginFormSubmit}
                    onUserChange={this.handleUserChange}
                    password={this.password}
                    user={this.user}
                />
            );
        }

        if (this.resetFormVisible) {
            return (
                <ResetForm
                    loading={userStore.loading}
                    onChangeForm={this.handleChangeToLoginForm}
                    onSubmit={this.handleResetFormSubmit}
                    onUserChange={this.handleUserChange}
                    success={userStore.resetSuccess}
                    user={this.user}
                />
            );
        }
    }

    renderBackLink() {
        const {backLink, initialized} = this.props;

        if (!initialized) {
            return null;
        }

        return (
            <a className={loginStyles.backLink} href={backLink}>
                <Icon className={loginStyles.backLinkIcon} name={BACK_LINK_ARROW_LEFT_ICON} />
                {translate('sulu_admin.back_to_website')}
            </a>
        );
    }

    render() {
        return (
            <div className={loginStyles.login}>
                <div className={loginStyles.loginContainer}>
                    <div className={loginStyles.formContainer}>
                        <div className={loginStyles.logoContainer}>
                            <Icon name="su-sulu" />
                        </div>
                        {this.renderForm()}
                    </div>
                    <div className={loginStyles.backLinkContainer}>
                        {this.renderBackLink()}
                    </div>
                </div>
            </div>
        );
    }
}
