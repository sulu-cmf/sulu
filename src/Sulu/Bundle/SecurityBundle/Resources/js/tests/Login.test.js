/* eslint-disable flowtype/require-valid-file-annotation */
import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../Login';

test('Render login', () => {
    const login = renderer.create(<Login />);
    expect(login).toMatchSnapshot();
});
