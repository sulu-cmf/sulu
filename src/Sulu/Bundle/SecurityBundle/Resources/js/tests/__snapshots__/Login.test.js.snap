// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render login 1`] = `
<div>
  Login
</div>
`;
